src/coprocessor/dag/expr/builtin_string.rs

```diff
@@ -312,6 +312,47 @@ impl ScalarFunc {
         self.children[i as usize].eval_string(ctx, row)
     }
 
+    #[inline]
+    pub fn field_int(&self, ctx: &mut EvalContext, row: &[Datum]) -> Result<Option<i64>> {
+        // Pattern matching made explicit here.
+        // As per the MySQL doc, if the first argument is NULL, this function always returns 0.
+        let s = match self.children[0].eval_string(ctx, row) {
+            Err(e) => return Err(e),
+            Ok(None) => return Ok(Some(0)),
+            Ok(Some(res)) => res,
+        };
+
+        let mut i = 1;
+        while i < self.children.len() as i64 {
```

Comment:
How about `self.children.iter().skip(1).enumerate()`?

Reply:
It seems hard to model this function using only combinators due to subtle handling of NULL. I used a for loop instead.